src/controllers/task.controller.ts

```diff
@@ -20,4 +20,62 @@ export const getTaskByStatus = async (req: Request, res: Response): Promise<void
     res.status(200).send(tasks);
   }
 
-}
\ No newline at end of file
+}
+
+export const addTask = async (req: Request, res: Response): Promise<void> => {
+    
+  const data: string = req.body?.task;
+
+  if (data === undefined){
+    res.status(400).send('Missing required field: task');
+  } else {
+    const newTaskID = await taskService.addTask(data);
+    res.status(200).send({ id: newTaskID.id });
+  }
+
+}
+
+export const deleteTaskByID = async (req: Request, res: Response): Promise<void> => {
+    
+  const id = req.params?.id;
+  try{
+    await taskService.deleteTaskByID(id);
+    res.status(200).send();
+  }catch(err){
+    res.status(400).send(err);
+  }
+  
+}
+
+export const deleteTaskByStatus = async (req: Request, res: Response): Promise<void> => {
+    
+  const status: string = req.params?.status;
+
+  if (status !== ETaskStatus.Complete && status !== ETaskStatus.Incomplete) {
+    res.status(400).send('Not a valid status');
+  } else {
+    await taskService.deleteTaskByStatus(status);
+    res.status(200).send();
+  }
+
+}
+
+export const updateTask = async (req: Request, res: Response): Promise<void> => {
+    
+  const id: string = req.params?.id;
+  const status: string = req.params?.status;
+
+  try{
+
+    if (status !== ETaskStatus.Complete && status !== ETaskStatus.Incomplete) {
```

Comment:
You can also write as below
```suggestion
    if (![ETaskStatus.Complete,ETaskStatus.Incomplete].some(t=>t === status)) 
    { 
      // throw exception as invalid 
    }
```